python/src/com/jetbrains/python/psi/types/PyModuleType.java

```diff
@@ -132,6 +109,103 @@ else if (location != null) {
     return null;
   }
 
+  private static void processImplicitPackageMembers(@NotNull PsiFileSystemItem anchor,
```

Comment:
This method has become quite long. Could we split it into parts? Some parts that I see are:

* Collecting the list of elements that may introduce implicit names
* Searching for implicit names in the collected imports
* Searching for implicit names in the importable qualified name of the location module